lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -34,6 +35,45 @@ PXR_NAMESPACE_USING_DIRECTIVE
 namespace MAYAUSD_NS_DEF {
 namespace ufe {
 
+namespace {
+// We will not use the value of UsdUtilsGetMaterialsScopeName() for the material scope.
+static const std::string kDefaultMaterialScopeName("mtl");
+
+#if (UFE_PREVIEW_VERSION_NUM >= 4010)
+bool connectShaderToMaterial(
+    Ufe::SceneItem::Ptr shaderItem,
+    UsdPrim             materialPrim,
+    const std::string&  nodeId)
+{
+    auto                 shaderUsdItem = std::dynamic_pointer_cast<UsdSceneItem>(shaderItem);
+    auto                 shaderPrim = UsdShadeShader(shaderUsdItem->prim());
+    UsdShadeOutput       materialOutput;
+    PXR_NS::SdrRegistry& registry = PXR_NS::SdrRegistry::GetInstance();
+    PXR_NS::SdrShaderNodeConstPtr shaderNodeDef
+        = registry.GetShaderNodeByIdentifier(TfToken(nodeId));
+    SdrShaderPropertyConstPtr shaderOutputDef;
+    if (shaderNodeDef->GetSourceType() == "glslfx") {
```

Comment:
Again, pointer used without checking for validity: are we guaranteed the node will always exists?